src/oidc/validate-token.ts

```diff
@@ -0,0 +1,83 @@
+import {decodeJwt} from "jose";
+
+const SUPPORTED_REDIRECT_URLS = [
+    "https%3A%2F%2Fhome.dev.account.gov.uk%2Fauth%2Fcallback",
+    "http%3A%2F%2Flocalhost%3A6001%2Fauth%2Fcallback",
+    "https%3A%2F%2Fhome.build.account.gov.uk%2Fauth%2Fcallback"
+];
+const ALLOWED_GRANT_TYPES = ['authorization_code'];
+
+export const getIssuerFromJWT = (token: string) => {
```

Comment:
```suggestion
export const getIssuerFromJWT = (token: string): string | undefined => {
```
Let's lean on the type system here to help us not make mistakes. This function returns either a string or `null` so let's define that; it'll force us to explicitly handle both cases when we call it. 

I know you've done that already in `validateClientIdMatches` here, but I'm thinking if we ever come back to change this in a few months time and have forgotten the context.

Reply:
Fixed, thanks